migrations/2018-08-20-00-migrate-to-create-member-and-verify.ts

```diff
@@ -0,0 +1,217 @@
+      }
+
+      const updatesToOperation = {
+        op_code: OperationType.CREATE_MEMBER,
+        "data.request_invite_from_member_id": inviterMemberId,
+        "data.to_uid": admin.firestore.FieldValue.delete()
+      };
+      console.log("=============================");
+      console.log("Updating operation", operation.id);
+      console.log("Operation was", operationData);
+      console.log("Update", updatesToOperation);
+      console.log("=============================");
+      if (!dryRun) {
+        await operation.ref.update(updatesToOperation);
+        ++updatedOperations;
+      }
+    }
+    if (operationData.op_code === OperationType.TRUST) {
+      const opCreatorUid = operationData.creator_uid;
+
+      const trusterMemberId =
+        opCreatorUid === null &&
+        ORIGINAL_GENESIS_TRUST_OPS.indexOf(operation.id) >= 0
+          ? GENESIS
+          : opCreatorUid;
+      const trustedMemberId = operationData.data.to_uid;
+
+      if (invitedBy[trustedMemberId] === trusterMemberId) {
+        // Check if this is a request-invite confirming Trust operation
+        // Check if the requisite "Verify" operation already exists.
+        // Allows this migration to be idempotent.
+        if (
+          !(await operationsCollection
+            .where("op_code", "==", OperationType.VERIFY)
+            .where("creator_uid", "==", trusterMemberId)
+            .where("data.to_uid", "==", trustedMemberId)
+            .get()).empty
+        ) {
+          continue;
+        }
+
+        const newVerifyOperation: Omit<VerifyOperation, "id"> = {
+          op_code: OperationType.VERIFY,
+          creator_uid: trusterMemberId,
+          created_at: new Date(operationData.created_at),
+          data: {
+            to_uid: trustedMemberId,
+            video_url: getInviteVideoUrlForMemberId(trustedMemberId)
+          }
+        };
+
+        console.log("=============================");
+        console.log(
+          "Creating a new verify operation for trust operation with id",
+          operation.id
+        );
+        console.log("Trust operation was", operationData);
+        console.log("Verify operation is", newVerifyOperation);
+        console.log("=============================");
+        if (!dryRun) {
+          const newVerifyOp = await operationsCollection.add(
+            newVerifyOperation
+          );
+          if (ORIGINAL_GENESIS_TRUST_OPS.indexOf(operation.id) >= 0) {
+            await operation.ref.delete();
+            newGenesisVerifyOperations.push(newVerifyOp.id);
+          }
+          ++createdOperations;
+        }
+      }
+    }
+  }
+  console.log(`Created ${createdOperations} operations.`);
+  console.log(`Updated ${updatedOperations} operations.`);
+  if (newGenesisVerifyOperations.length > 0) {
+    console.log("!!!!!! IMPORTANT !!!!!!!");
+    console.log(
+      "New GENESIS Verify operations were created, and the original GENESIS Trust operations were deleted."
```

Comment:
Only if not dry run, right?

Reply:
correct. this array should only be populated if dryRun is false.

Reply:
ding